### PR TITLE
JsonObject reports multiple mismatches

### PR DESCRIPTION
**Pull request by eshrubs**
In the current implementation only the first mismatch will be detected. This is
not desirable as one `assertThat` should show all mistmatches, as is done with 
similar hamcrest matches like `containsInAnyOrder`

### CODE REVIEW

File: jackson/src/main/java/com/spotify/hamcrest/jackson/IsJsonObject.java [hunk 1]
```diff
@@ -26,13 +26,16 @@
...
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 import java.util.function.Consumer;
 import org.hamcrest.Description;
 import org.hamcrest.Matcher;
 import org.hamcrest.StringDescription;
 
+
```
Comment (dflemstr) [inline jackson/src/main/java/com/spotify/hamcrest/jackson/IsJsonObject.java]: ?

File: jackson/src/test/java/com/spotify/hamcrest/jackson/IsJsonObjectTest.java [hunk 2]
```diff
@@ -84,10 +85,10 @@ public void testMismatch() throws Exception {
 
     assertThat(description.toString(), is(
         "{\n"
-        + "  ...\n"
-        + "  \"bar\": was a boolean node was <true>\n"
-        + "  ...\n"
-        + "}"
+            + "  ...\n"
```
Comment (dflemstr) [inline jackson/src/test/java/com/spotify/hamcrest/jackson/IsJsonObjectTest.java]: Would prefer to leave this indentation as is :-)

File: jackson/src/test/java/com/spotify/hamcrest/jackson/IsJsonObjectTest.java [hunk 3]
```diff
@@ -132,6 +133,63 @@ public void testMismatchType() throws Exception {
...
+        .put("baz", 3);
+
+    final StringDescription description = new StringDescription();
+    sut.describeMismatch(nestedMismatches, description);
+
+    assertThat(description.toString(), is(
+        "{\n"
+            + "  ...\n"
+            + "  \"foo\": was a number node was <-1>\n"
+            + "  ...\n"
```
Comment (dflemstr) [inline jackson/src/test/java/com/spotify/hamcrest/jackson/IsJsonObjectTest.java]: Nice, wanted this for a long time!

**Review by dflemstr — CHANGES_REQUESTED** [review]

**Comment by eshrubs** [comment]
@dflemstr Go nuts!

**Comment by eshrubs** [comment]
@dflemstr Go nuts!

**Comment by eshrubs** [comment]
@davidxia @mattnworb Poke!

**Review by mattnworb — APPROVED** [review]
Neat, I didn't realize standard matchers like `containsInAnyOrder` did this. I've definitely written a bunch of one off matchers that only report the first mismatch. LGTM!

**Pull request merged**